drivers/hmis_csv_importer/app/jobs/hmis_csv_importer/mark_expired_job.rb

```diff
@@ -0,0 +1,55 @@
+module HmisCsvImporter
+  class MarkExpiredJob < ApplicationJob
+    def perform(now: DateTime.current, recent_imports_to_keep: 10, keep_deleted_records_for: 2.weeks)
+      @now = now.to_datetime
+      @expire_deleted_records_before = now - keep_deleted_records_for
+      @recent_imports_to_keep = recent_imports_to_keep
+
+      total = 0
+      models.each do |model|
+        # FIXME, should include select clause to improve performance
+        candidates_for_expiration = model.
+          where.not(loader_id: recent_import_ids).
+          preload(:destination_record_with_deleted)
+
+        puts "#{model.name} total: #{candidates_for_expiration.size}"
+        candidates_for_expiration.in_batches(of: 1_000) do |batch|
+          expired_ids = []
+
+          batch.each do |record|
+            destination = record.destination_record_with_deleted
+            expired_ids.push(record.id) if expired?(record, destination)
+          end
+          batch.where.not(id: expired_ids).update_all(expired: false)
+          model.where(id: expired_ids).update_all(expired: true)
+          total += expired_ids.size
+        end
+        puts "#{model.name} expired: #{candidates_for_expiration.where(expired: true).size}"
+      end
+
+      total
+    end
+
+    protected
+
+    def recent_import_ids
+      @recent_import_ids ||= HmisCsvImporter::Loader::LoaderLog.order(created_at: :desc).limit(@recent_imports_to_keep).pluck(:id)
```

Comment:
There's some complexity here we need to account for.  These will need to be per-data source, and it will need to multiply the `@recent_imports_to_keep` by `HmisImportConfig.file_count` if there is a config for that data source.